jvm-libs/linea/metrics/micrometer/src/main/kotlin/net/consensys/linea/metrics/micrometer/MicrometerMetricsFacade.kt

```diff
@@ -37,26 +38,28 @@ class MicrometerMetricsFacade(
 
   init {
     if (metricsPrefix != null) requireValidMicrometerName(metricsPrefix)
+    defaultTags.forEach { requireValidMicrometerName(it.key) }
   }
 
-  private fun metricHandle(category: LineaMetricsCategory?, metricName: String): String {
+  private fun metricHandle(category: MetricsCategory?, metricName: String): String {
     val prefixName = if (metricsPrefix == null) "" else "$metricsPrefix."
-    val categoryName = if (category == null) "" else "$category."
+    val categoryName = if (category == null) "" else "${category.toValidMicrometerName()}."
     return "$prefixName$categoryName$metricName"
   }
 
   override fun createGauge(
-    category: LineaMetricsCategory?,
+    category: MetricsCategory?,
     name: String,
     description: String,
     measurementSupplier: Supplier<Number>,
     tags: List<Tag>,
   ) {
-    if (category != null) requireValidMicrometerName(category.toString())
+    if (category != null) requireValidMicrometerName(category.toValidMicrometerName())
     requireValidMicrometerName(name)
     val builder = Gauge.builder(metricHandle(category, name), measurementSupplier)
-    if (tags.isNotEmpty()) {
-      val flatTags = tags.flatMap {
+    if (tags.isNotEmpty() || defaultTags.isNotEmpty()) {
```

Comment:
I think we should reuse this piece of logic for other kinds of metrics

Reply:
I meant that we should reuse this piece of code. Would something like this be worth it?

```
private inline fun <T> applyTagsIfNotEmpty(
    builder: T,
    tags: List<Tag>,
    crossinline tagsApplier: T.(Array<out String>) -> T
  ): T = flattenTags(tags).takeIf { it.isNotEmpty() }?.let { builder.tagsApplier(it.toTypedArray()) } ?: builder
```

And you call it like 
```
applyTagsIfNotEmpty(builder, tags) { tags(*it) }
```